### ISSUE TITLE
Cannot clone LinkedHashMap - StackOverflowError

### ISSUE THREAD

**Issue by epaaj** [kostaskougios/cloning#53]
When trying to clone a LinkedHashMap you get StackOverflowError.
It doesn't matter if it is empty or not.
```
Cloner cloner = new Cloner();
LinkedHashMap<Integer, Integer> test = new LinkedHashMap<>();
cloner.deepClone(test);
```

**Comment by kostaskougios** [comment]
Works ok for me, are you using version 1.9.2?

For large maps it fails with a stack overflow error. I've fixed this for 1.9.3 (which is not yet deployed, will try to deploy soon)

**Comment by epaaj** [comment]
You're right, it does work. Maybe I broke it while messing with the code. Sorry about that. I'm looking forward to 1.9.3 being released!
